packages/swapper/src/types.ts

```diff
@@ -229,12 +272,25 @@ export type MultiHopTradeQuote = TradeQuoteBase & {
   steps: MultiHopTradeQuoteSteps
 }
 
+export type SingleHopTradeRate = TradeRateBase & {
+  steps: SingleHopTradeQuoteSteps
+}
+export type MultiHopTradeRate = TradeRateBase & {
+  steps: MultiHopTradeQuoteSteps
+}
+
 // Note: don't try to do TradeQuote = SingleHopTradeQuote | MultiHopTradeQuote here, which would be cleaner but you'll have type errors such as
 // "An interface can only extend an object type or intersection of object types with statically known members."
 export type TradeQuote = TradeQuoteBase & {
   steps: SingleHopTradeQuoteSteps | MultiHopTradeQuoteSteps
 }
 
+export type TradeRate = TradeRateBase & {
+  steps: SingleHopTradeQuoteSteps | MultiHopTradeQuoteSteps
+}
+
+export type TradeQuoteOrRate = TradeQuote | TradeRate
```

Comment:
Not sure there would be much surface area change since in effect, this is really a type rename i.e lsp renamed `TradeQuote` to `TradeQuoteOrRate`.
The main benefit of this *currently* is having a clear-defined type which we can use a `isTradeRate`  type guard for, see an example of consumption here: https://github.com/shapeshift/web/blob/7e2a8b56341d91b37c3148e53f9bfdfe5775bec9/src/lib/tradeExecution.ts#L73-L75

Though keep in mind this PR was originally written with the intent of having a second PR where we do clear rates and quotes firing, i.e rates at input time, then when at confirm time with a wallet, a second quote which is a trade quote, see this copypasta from engineering DM below.

As mentioned in that message, there are many implications to that and given this already does the do functionally vs. keeping rates in memory when connecting a wallet, I'm not sure if that's realistic for us to go with this functional split in the end. Although I believe there's nothing wrong keeping this, as this improves naming, does a clear types distinction, and makes us safer? 

```
On the finish line for https://github.com/shapeshift/web/pull/7939, currently regression testing it before opening and would like more brains on this for Monday producty things/Wednesday gangles.

tl;dr:

We don't have to proper split between quotes and rates, a quote is just an augmented rate with receiveAddress (types, including input types have been revamped toreflect that)
Most of the magic happens with a combination of types and runtime safety, as well as input referential invalidation:
Types and runtime safety ensure that we don't execute a trade rate, but can get one without addresses (from/to), with checks at getUnsignedTx time to ensure we don't build a payload without addies/bip44 stuff
Referential invalidation ensures that as soon as you connect a wallet, quotes refetch

In the end, this is much more straightforward as opposed to properly splitting between quotes (pre-execution time) and rates (input time), as all swappers have many implications re: keeping quotes gotten at input time at the moment. If we did this, we would also have to ensure that we don't end up with too big of rates variations, and reflect said variations to users if any.

Works pretty nicely as-is, and input is persisted, however, as per design, quotes refetch on wallet connection, which is actually part of the safety model here. This means that quotes aren't, and as a result, nor is the selected quote

https://jam.dev/c/b7a20304-e871-4ea8-a772-3b30ba48d756 
https://jam.dev/c/e7c07d97-ce8f-4c34-8578-f86296fe2dc8

The q. here is whether we really want to go with another swapper architectural on top of this PR, to shave a few seconds getting quotes again on wallet connect which is going to be a ~3 of sorts and a high-risk once again, or if we're happy with this flow. 
```

Reply:
After discussing this one out in DM, removed the split for `TradeQuote` (`TradeQuoteOrRate`), though kept for `TradeQuoteInput` for safety: https://github.com/shapeshift/web/pull/7939/commits/299c8a403c81dba47853c027eb7b1b63a891754c